hal/src/sercom/i2c/config.rs

```diff
@@ -253,31 +258,33 @@ impl<P: PadSet> Config<P> {
 pub trait AnyConfig: Is<Type = SpecificConfig<Self>> {
     type Sercom: Sercom;
     type Pads: PadSet<Sercom = Self::Sercom>;
+    type Mode: I2cMode; // default to super::Master when associated_type_defaults is stable
```

Comment:
I don't think `AnyConfig` is meant to be implemented outside of the HAL anyways.

Reply:
Oh good point, `Sercom` and `Padset` are sealed so it's de facto sealed anyway. I'll make that explicit just so it's also clear in the docs.